.github/workflows/build-linux.yml

```
@@ -0,0 +1,21 @@
+name: build-linux
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3 
+      - run: sudo apt install -y apt-utils build-essential wget
+      - run: sudo apt install -y qt5-qmake qt5-qmake-bin qt5-assistant qtbase5-dev qtmultimedia5-dev
+      - run: sudo apt install -y libqt5charts5 libqt5charts5-dev  libqt5multimedia*
+      - run: sudo apt install -y libqt5datavisualization5-dev libqt5datavisualization5
+      - run: sudo apt install -y libopencv-core-dev libopencv-core4.5d libopencv-dev
+      - run: sudo apt install -y libqwt-qt5-6 libqwt-qt5-dev
+      - run: sudo apt install -y libarmadillo-dev libarmadillo10
+      - run: qmake
+      - run: make -j4
```

Why break up the installation of the dependencies into separate commands? Also, you can combine multiple commands into a single `run`:

```suggestion
      - run: |
          sudo apt install -y apt-utils build-essential wget qt5-qmake qt5-qmake-bin qt5-assistant qtbase5-dev qtmultimedia5-dev libqt5charts5 libqt5charts5-dev  libqt5multimedia* libqt5datavisualization5-dev libqt5datavisualization5 libopencv-core-dev libopencv-core4.5d libopencv-dev libqwt-qt5-6 libqwt-qt5-dev libarmadillo-dev libarmadillo10
          qmake
          make -j4
```

---

I don't think it makes much difference either way.  I think the older way is slightly more readable I suppose?

---

@gr5 No, it ultimately doesn't matter, I just don't get why you would install dependencies a few at a time - it's not a pattern that I've seen before :shrug:. If for example you wanted to install these dependencies on your local machine, you'd have to copy/paste/run 7 lines instead of 1. Mostly I like to avoid quirks in the code base, it makes it harder for people to read and understand the intent of the original author.

---

hi peyton,

the "apt install" order mostly matched the compile failures on the initial linux compile

clay

---

yes maybe we should have 3 run : 
apt-install
qmake
make
I agree it makes not much sense to separate apt-installs
And having 3 separate run will increase readability in output (see https://github.com/atsju/DFTFringe/actions/runs/4186601870/jobs/7255312510)